server/modules/activityLog/repository/activityLogRepository.js

```diff
@@ -23,8 +25,12 @@ export const insertMany = async (user, surveyId, activities, client) =>
   ])
 
 //===== READ
-export const fetch = async (surveyId, activityTypes = null, offset = 0, limit = 30, client = db) =>
-  await client.map(`
+export const fetch = async (surveyInfo, activityTypes = null, offset = 0, limit = 30, client = db) => {
+  const surveyUuid = Survey.getUuid(surveyInfo)
+  const surveyId = Survey.getIdSurveyInfo(surveyInfo)
+  const schema = getSurveyDBSchema(surveyId)
```

Comment:
can't you pass just the surveyId as it was before? to get the surveyUuid you can join with survey table? it's just weird to pass the surveyInfo to the repository... isn't it?

Reply:
you have always the surveyInfo when you call ActiviityLogRepository.fetch, so you can save a join, but I'll change it if you want